optaplanner-benchmark/src/test/java/org/optaplanner/benchmark/impl/measurement/ScoreDifferencePercentageTest.java

```diff
@@ -0,0 +1,192 @@
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+
+import static org.junit.Assert.assertEquals;
+
+public class ScoreDifferencePercentageTest {
+
+    @Test(expected = IllegalStateException.class)
+    public void calculateScoreDifferencePercentageException() {
+        int[] hardScore1 = {1, 2, 3};
+        int[] softScore1 = {4, 5};
+        int[] hardScore2 = {1, 2};
+        int[] softScore2 = {4, 5};
+        BendableScore score1 = BendableScore.valueOf(hardScore1, softScore1);
+        BendableScore score2 = BendableScore.valueOf(hardScore2, softScore2);
+        ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+    }
+
+    @Test
+    public void calculateScoreDifferencePercentage() {
+        double tolerance = 0.00001;
+        SimpleScore score1 = SimpleScore.parseScore("-100");
+        SimpleScore score2 = SimpleScore.parseScore("-100");
+        ScoreDifferencePercentage scoreDifferencePercentage =
+                ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+        assertEquals(0.0, scoreDifferencePercentage.getPercentageLevels()[0], tolerance);
+
+        score1 = SimpleScore.parseScore("100");
+        score2 = SimpleScore.parseScore("100");
+        scoreDifferencePercentage =
+                ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+        assertEquals(0.0, scoreDifferencePercentage.getPercentageLevels()[0], tolerance);
+
+        score1 = SimpleScore.parseScore("-100");
+        score2 = SimpleScore.parseScore("-10");
+        scoreDifferencePercentage =
+                ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+        assertEquals(0.9, scoreDifferencePercentage.getPercentageLevels()[0], tolerance);
+
+        score1 = SimpleScore.parseScore("100");
+        score2 = SimpleScore.parseScore("10");
+        scoreDifferencePercentage =
+                ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+        assertEquals(-0.9, scoreDifferencePercentage.getPercentageLevels()[0], tolerance);
+
+        score1 = SimpleScore.parseScore("-100");
+        score2 = SimpleScore.parseScore("-1");
+        scoreDifferencePercentage =
+                ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+        assertEquals(0.99, scoreDifferencePercentage.getPercentageLevels()[0], tolerance);
+
+        score1 = SimpleScore.parseScore("100");
+        score2 = SimpleScore.parseScore("1");
+        scoreDifferencePercentage =
+                ScoreDifferencePercentage.calculateScoreDifferencePercentage(score1, score2);
+        assertEquals(-0.99, scoreDifferencePercentage.getPercentageLevels()[0], tolerance);
+
+        HardSoftScore hardSoftScore1 = HardSoftScore.parseScore("-100hard/-1soft");
```

Comment:
Prefer HardSoftScore.valueOf(-100, 1) in future PR's. Faster performance for tests (unnoticably, but still).